pkg/generate/source/detector.go

```diff
@@ -29,70 +29,67 @@ var DefaultDetectors = Detectors{
 	DetectPerl,
 	DetectScala,
 	DetectDotNet,
-	DetectLiteralDotNet,
 }
 
 // DetectRuby detects Ruby source
-func DetectRuby(dir string) (*Info, bool) {
+func DetectRuby(dir string) *Info {
 	return detect("ruby", dir, "Gemfile", "Rakefile", "config.ru")
 }
 
 // DetectJava detects Java source
-func DetectJava(dir string) (*Info, bool) {
+func DetectJava(dir string) *Info {
 	return detect("jee", dir, "pom.xml")
 }
 
 // DetectNodeJS detects NodeJS source
-func DetectNodeJS(dir string) (*Info, bool) {
+func DetectNodeJS(dir string) *Info {
 	return detect("nodejs", dir, "app.json", "package.json")
 }
 
 // DetectPHP detects PHP source
-func DetectPHP(dir string) (*Info, bool) {
+func DetectPHP(dir string) *Info {
 	return detect("php", dir, "index.php", "composer.json")
 }
 
 // DetectPython detects Python source
-func DetectPython(dir string) (*Info, bool) {
+func DetectPython(dir string) *Info {
 	return detect("python", dir, "requirements.txt", "setup.py")
 }
 
 // DetectPerl detects Perl source
-func DetectPerl(dir string) (*Info, bool) {
+func DetectPerl(dir string) *Info {
 	return detect("perl", dir, "index.pl", "cpanfile")
 }
 
 // DetectScala detects Scala source
-func DetectScala(dir string) (*Info, bool) {
+func DetectScala(dir string) *Info {
 	return detect("scala", dir, "build.sbt")
 }
 
 // DetectDotNet detects .NET source and matches it to a dotnet supported annotatin or dotnet imagestream name
-func DetectDotNet(dir string) (*Info, bool) {
-	return detect("dotnet", dir, "project.json")
-}
-
-// DetectLiteralDotNet detects .NET source and matches it to a .net supported annotation
-func DetectLiteralDotNet(dir string) (*Info, bool) {
-	return detect(".net", dir, "project.json")
+func DetectDotNet(dir string) *Info {
+	info := detect("dotnet", dir, "project.json")
+	if info == nil {
+		info = detectGlob("dotnet", dir, "*.csproj")
+	}
+	return info
 }
 
 // detect returns an Info object with the given platform if the source at dir contains any of the argument files
-func detect(platform string, dir string, files ...string) (*Info, bool) {
-	if filesPresent(dir, files) {
-		return &Info{
-			Platform: platform,
-		}, true
+func detect(platform string, dir string, files ...string) *Info {
+	for _, f := range files {
+		if _, err := os.Stat(filepath.Join(dir, f)); err == nil {
+			return &Info{Platform: platform}
+		}
 	}
-	return nil, false
+	return nil
 }
 
-func filesPresent(dir string, files []string) bool {
-	for _, f := range files {
-		_, err := os.Stat(filepath.Join(dir, f))
-		if err == nil {
-			return true
+func detectGlob(platform string, dir string, globs ...string) *Info {
+	for _, g := range globs {
+		if matches, _ := filepath.Glob(filepath.Join(dir, g)); len(matches) > 0 {
```

Comment:
can we not just do globbing in the existing detect function, regardless of whether the passed pattern contains a * ?